module-4/backend/week_four.md

```diff
@@ -1,20 +1,63 @@
 ## Week Four - Module 4 Recap
+### Lauren Oliveri
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+* What's the purpose of a project management tool?
 
-Note: When you're done, submit a PR. 
+Project Management tools allow teams to organize their project priorities, sprint features, and bug fixes. It can also facilitate tracking of story assignments, sprint progress, and help to estimate work load. 
 
-* What's the purpose of a project management tool?
 * What's your favorite project management tool?
+
+I really like Pivotal Tracker. I know that to many in the class, it seems a little archaic, but I like the sprint/release features, epic feature, and velocity tracking. These aren't things that seem completely helpful when working on small-scale projects at Turing, but I love the way they work on big-sized projects. I've had to use it while dabbling in product management at my old company and found it to be really robust.
+
 * Why do we create staging environments?
+
+Staging environments allow developers to have a place to test their features in a production-like environment before the environment is actually deployed to production and its users. 
+
 * What are the characteristics of a good README (in your opinion)?
+
+* Project configuration and installation instructions (clone this, bundle that, etc.) 
+* Contact information for the distributor or programmer
+* Common issues and troubleshooting steps
+* Known bugs
+* Instructions to submit bugs, feature requests, submit patches, join mailing list, get announcements, or join the user or dev community in other forms
+* Credits and Acknowledgements
+* Copyright and licensing information
+
 * What's one main improvement you're going to make to your code regarding accessibility issues?
+
+I will definitely make sure to be using links and buttons that are "tab-able", it's definitely something I didn't do in Quantified Self, and it can easily be implemented using the tabindex if I don't use an <a> or <button> tag.
+
 * What are some basic security concerns to be aware of when building applications?
+
+* SQL and JavaScript Injection - when a user injects SQL and JS into user inputs, so a best practice is never trust user input and interpolate directly - best to strip out SQL before sending to the DB and any <script> tags in post requests from users.
+* Click Jacking - employed on malicious websites to make you do things on other websites - common prevention measure is to prevent the browser from allowing your site to be loaded in an iframe.
+
 * Why is continuous integration helpful/important?
+
+Continuous Integration allows us to create an environment purely for testing. I've seen it be immensely helpful at my old company where Pull Requests automatically ran the CI test suite and were "unmergeable" unless the code passed all tests.
```

Comment:
Along with an environment purely for testing, the key is that it runs the tests relative to a production environment, so it can save us from differences between development and production.